component/attr_cache/attr_cache.go

```diff
@@ -725,10 +670,21 @@ func (ac *AttrCache) RenameFile(options internal.RenameFileOptions) error {
 		ac.cacheLock.RLock()
 		defer ac.cacheLock.RUnlock()
 		// TODO: Can we just copy over the attributes from the source to the destination so we don't have to invalidate?
-		ac.deletePath(options.Src, renameTime)
-		ac.invalidatePath(options.Dst)
-	}
 
+		toBeDeleted, err := ac.cacheMap.get(options.Src)
+		if err != nil {
+			log.Err("could not find attr cache item due to following error: ", err)
+		} else {
+			toBeDeleted.markDeleted(renameTime)
+		}
+
+		toBeInvalid, err := ac.cacheMap.get(options.Dst)
```

Comment:
If you are keeping renamed cache entries, like you've decided to do in `renameCachedDirectory()`, then we should copy the entry over instead of invalidating the destination.
Maybe you can make a new function that copies the cache entry over and use the same function here and in `renameCachedDirectory`.

Reply:
it looks like renameCachedDirectoryHelper() already is copying the data over to the destination just fine for us. maybe I should just call renameCachedDirectory in this function instead of invalidating?